### PR TITLE
[Form] Minor fixes in docs and cs

### PR DESCRIPTION
**Pull request by HeahDude**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | ~
| License       | MIT
| Doc PR        | ~

### CODE REVIEW

File: src/Symfony/Component/Form/Util/OrderedHashMapIterator.php [hunk 1]
```diff
@@ -56,17 +56,15 @@ class OrderedHashMapIterator implements \Iterator
...
 
     /**
-     * Creates a new iterator.
-     *
      * @param array $elements       The elements of the map, indexed by their
      *                              keys
      * @param array $orderedKeys    The keys of the map in the order in which
      *                              they should be iterated
-     * @param array $managedCursors An array from which to reference the
+     * @param array $managedCursors an array from which to reference the
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]: please revert fabbot patch, these are false positives :)

Comment (HeahDude) [reply src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]: Arf, I was wondering, thanks!

File: src/Symfony/Component/Form/Util/OrderedHashMapIterator.php [hunk 2]
```diff
@@ -66,7 +64,7 @@ class OrderedHashMapIterator implements \Iterator
      *                              iterator's cursor as long as it is alive.
      *                              This array is managed by the corresponding
      *                              {@link OrderedHashMap} instance to support
-     *                              recognizing the deletion of elements.
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]: should be kept also :)

**Review by nicolas-grekas — APPROVED** [review]
(fabbot failure is false positive)

File: src/Symfony/Component/Form/Util/OrderedHashMapIterator.php (hunk 2 above)
Comment (HeahDude) [reply src/Symfony/Component/Form/Util/OrderedHashMapIterator.php]: Ok, done!

File: src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php [hunk 3]
```diff
@@ -73,16 +73,17 @@ public function mapFormsToData($forms, &$data)
             // Write-back is disabled if the form is not synchronized (transformation failed),
             // if the form was not submitted and if the form is disabled (modification not allowed)
             if (null !== $propertyPath && $config->getMapped() && $form->isSubmitted() && $form->isSynchronized() && !$form->isDisabled()) {
+                $propertyValue = $form->getData();
                 // If the field is of type DateTime and the data is the same skip the update to
                 // keep the original object hash
-                if ($form->getData() instanceof \DateTime && $form->getData() == $this->propertyAccessor->getValue($data, $propertyPath)) {
+                if (($propertyValue instanceof \DateTime || $propertyValue instanceof \DateTimeInterface) && $propertyValue == $this->propertyAccessor->getValue($data, $propertyPath)) {
```
Comment (fabpot) [inline src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: Looks like a bug fix?

Comment (HeahDude) [reply src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: Indeed, do you prefer another commit with the test?

Comment (fabpot) [reply src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: yes please (especially as the commit message should explain the fix)

Comment (HeahDude) [reply src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php]: Extracted in #29185.

File: src/Symfony/Component/Form/Form.php [hunk 4]
```diff
@@ -341,8 +343,8 @@ public function setData($modelData)
             $modelData = $event->getData();
         }
 
-        // Treat data as strings unless a value transformer exists
-        if (!$this->config->getViewTransformers() && !$this->config->getModelTransformers() && is_scalar($modelData)) {
+        // Treat data as strings unless a transformer exists
+        if (is_scalar($modelData) && !$this->config->getViewTransformers() && !$this->config->getModelTransformers()) {
```
[the remark this answers is not in the document]
Comment (xabbuh) [reply src/Symfony/Component/Form/Form.php]: My bad, I simply missed that we already had the `is_scalar()` check before.

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @HeahDude.

**Pull request merged**